src/main/java/org/broadinstitute/hellbender/tools/validation/CompareBaseQualities.java

```diff
@@ -0,0 +1,85 @@
+package org.broadinstitute.hellbender.tools.validation;
+
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.SamReader;
+import htsjdk.samtools.SamReaderFactory;
+import htsjdk.samtools.SecondaryOrSupplementarySkippingIterator;
+import htsjdk.samtools.util.CloserUtil;
+import htsjdk.samtools.util.IOUtil;
+import org.broadinstitute.hellbender.cmdline.*;
+import org.broadinstitute.hellbender.cmdline.programgroups.ReadProgramGroup;
+import org.broadinstitute.hellbender.engine.ProgressMeter;
+import org.broadinstitute.hellbender.exceptions.UserException;
+
+import java.io.File;
+import java.util.List;
+
+@CommandLineProgramProperties(
+        summary = "USAGE: CompareBaseQualities <SAMFile1> <SAMFile2>\n" +
+                "Compares the base qualities of two input SAM/BAM/CRAM files. The files must be sorted exactly the same name.",
+        oneLineSummary = "Compares base qualities of two input SAM/BAM/CRAM files",
+        programGroup = ReadProgramGroup.class
+)
+public final class CompareBaseQualities extends PicardCommandLineProgram {
+    @PositionalArguments(minElements = 2, maxElements = 2)
+    public List<File> samFiles;
+
+    @Argument(doc="Whether to allow  differences in read order", shortName = "allow_different_read_order", fullName = "allow_different_read_order")
+    private boolean allowDifferentReadOrder= false;
```

Comment:
Hi David,

Maybe this discussion might be helpful:

http://sourceforge.net/p/samtools/mailman/message/29208613/

And Picard provides the following method:

https://github.com/broadinstitute/picard/blob/b7a1335ffaf7400573145c3ac26de10462744c8b/src/java/picard/sam/CompareSAMs.java#L291-L313

Hope it helps,
~p